### PR TITLE
Adjusted testcases for better pipe plugin

### PR DESCRIPTION
**Pull request by veewee**
See https://github.com/php-standard-library/psalm-plugin/pull/8

The static-analysis tests are slightly adjusted to ensure better coverage.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -57,5 +57,11 @@
     },
     "suggest": {
         "php-standard-library/psalm-plugin": "Psalm integration"
-    }
+    },
+    "repositories": [
+        {
+            "type": "git",
```
Comment (veewee) [inline composer.json]: Will revert this if psalm plugin is merged.

File: tests/static-analysis/Fun/pipe.php [hunk 2]
```diff
@@ -46,11 +44,12 @@ function test_variadic_and_default_params(): void
  * @see https://github.com/vimeo/psalm/issues/7244
  *
  * @psalm-suppress InvalidArgument
+ * @psalm-suppress NoValue - Resolves into "\Closure(never): never" because of the issue linked above.
  */
-function test_empty_pipe(): void
+function test_empty_pipe(): string
```
Comment (veewee) [inline tests/static-analysis/Fun/pipe.php:49]: This is still an issue....  Currently no way around this.

https://github.com/vimeo/psalm/issues/7244

File: src/Psl/Network/ServerInterface.php [hunk 3]
```diff
@@ -27,7 +27,7 @@ public function nextConnection(): SocketInterface;
      *
      * @throws Exception\RuntimeException In case failed to accept incoming connection.
      *
-     * @return Generator<null, SocketInterface, void, void>
+     * @return Generator<null, SocketInterface, void, void|null>
```
Comment (azjezz) [inline src/Psl/Network/ServerInterface.php]: can we make this `null` as well ( same for the rest ).

**Comment by azjezz** [comment]
plugin is merged.

File: src/Psl/Network/ServerInterface.php (hunk 3 above)
Comment (veewee) [reply src/Psl/Network/ServerInterface.php]: Done

File: composer.json (hunk 1 above)
Comment (veewee) [reply composer.json]: Shall I release psalm-plugin as 2.2 (new feature) and after that this PSL as 2.3.1 (no changes besides tests)?

Comment (azjezz) [reply composer.json]: yes

**Review by azjezz — DISMISSED** [review]

**Review by azjezz — APPROVED** [review]

**Comment by azjezz** [comment]
let's not release this, as i don't see any benefits down stream, the require-dev constraint is not enforced for users.

**Comment by veewee** [comment]
Was thinking the same this morning. good!

**Pull request merged**